oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettings.java

```diff
@@ -93,6 +93,22 @@
 
     private String[] classNamesIgnoredInCallTrace = new String[] {};
 
+
+    private static final String OAK_QUERY_LENGTH_WARN_LIMIT = "oak.query.length.warn.limit";
+    private static final String OAK_QUERY_LENGTH_ERROR_LIMIT = "oak.query.length.error.limit";
+
+    private final long queryLengthWarnLimit = Long.getLong(OAK_QUERY_LENGTH_WARN_LIMIT, 102400); // 100 KB
+    private final long queryLengthErrorLimit = Long.getLong(OAK_QUERY_LENGTH_ERROR_LIMIT, 1048576); //1MB (1024KB)
```

Comment:
We currently want to see if there are warnings, and not throw errors, so 100 * 1024 * 1024 would be better I think (100 MiB). BTW did you know that newer Java versions supports "_" in numbers, so 1_000_000 would be 1 million. But I think using 100 * 1024 is OK (serves the same purpose).